src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathNodeHelper.cs

```diff
@@ -117,6 +118,7 @@ public static bool GetContentChild(ref XPathNode[] pageNode, ref int idxNode)
                 {
                     idx = page[idx].GetSibling(out page);
                     Debug.Assert(idx != 0);
+                    Debug.Assert(page != null);
```

Comment:
consider changing GetSibling/GetParent if they always return non-null

Reply:
I think the rule for GetSibling/GetParent and similar methods is that they are null when the return value is 0.
https://github.com/dotnet/runtime/blob/ad90f7ab81b99fab7ca4891c3943021827e66315/src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathNode.cs#L184-L187